### PR TITLE
Document removal fix from prompt studio

### PR DESCRIPTION
**Pull request by johnyrahul**
## What

- Document removal  was failing when shared user tries to delete the document.
- Use the proper HTTP method for deleting the document. 

## Why

- Following proper standards

## How

- Re-structured the Prompt studio  API used.

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

- Validated  by deleting the document by owner and shared user.

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/prompt_studio/prompt_studio_core/views.py [hunk 1]
```diff
@@ -444,11 +445,9 @@ def upload_for_ide(self, request: HttpRequest, pk: Any = None) -> Response:
         return Response({"data": documents})
 
     @action(detail=True, methods=["delete"])
-    def delete_for_ide(self, request: HttpRequest, pk: Any = None) -> Response:
+    def delete_for_ide(self, request: HttpRequest, pk: uuid) -> Response:
         custom_tool = self.get_object()
-        serializer = FileInfoIdeSerializer(data=request.GET)
-        serializer.is_valid(raise_exception=True)
```
Comment (Deepak-Kesavan) [inline backend/prompt_studio/prompt_studio_core/views.py:450]: @johnyrahul Why is the serializer validation removed?

Comment (johnyrahul) [reply backend/prompt_studio/prompt_studio_core/views.py:450]: @Deepak-Kesavan  the above serializer is expecting type as well. For this particular API it's only expecting document ID inside the body. I will go ahead and create a new serializer

Comment (johnyrahul) [reply backend/prompt_studio/prompt_studio_core/views.py:450]: @Deepak-Kesavan  addressed

**Review by Deepak-Kesavan — APPROVED** [review]

**Review by mohamed-siddhiq — APPROVED** [review]
LGTM

**Pull request merged**